### ISSUE TITLE
Indefinite loop when working with airserv

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1582]
**Reported by anonymous on 20 May 2015 10:47 UTC**

I've came across a bug which was introduced in the latest 1.2-rc2.
in osdep/network.c the function net_get_nopacket is stuck in indefinite loop.

The way to reproduce it:
1. Open a connection using wi_open to an airserv server
2. Kill the server (to kill the connection between the client and the server)
3. Invoke wi_get_monitor for example, to trigger the flow to net_cmd -> net_get_nopacket, which will in turn never return.

The reason is that net_get returns -1 when the connection is lost, which is not handled by the loop.

As you can see by the diff from rc1 to the trunk below, the -1 was handled properly:
```c
     while (1) {
           l = sizeof(buf);
           c = net_get(pn->pn_s, buf, &l);
214,221c214,219
<         if (c<0)
<             return c;
<               if (c == NET_PACKET)
<         {
<             net_enque(pn, buf, l);
<                       continue;
<         }
<         break;
---
> 
>               if (c != NET_PACKET && c > 0)
>                       break;
> 
>                 if(c > 0)
>                     net_enque(pn, buf, l); 
}
```

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 20 May 2015 10:47 UTC**

Sorry for the confusion!
I've just noticed that this was a local patch and it was NOT in the rc1!
Nevertheless, the patch should be applied in order to fix the indefinite loop.

**Comment by aircrack-ng** [comment]
**Comment by misterx on 20 May 2015 10:47 UTC**

Fixed in aircrack-ng/aircrack-ng@34136473665e9d6e284b9f32c03ba0307ab2b9e4.
